### PR TITLE
#156005187 Handle configuration changes

### PR DESCRIPTION
**Pull request by leonardnjura**
#### What does this PR do?
- Integrates recycler view adapter 
- Maintains screen view and data on orientation changes

#### Description of Task to be completed?
- Create Github users adapter
- Handle `onResume(),  onSaveInstanceState()` and `onRestoreInstanceState()` methods
- Fetch Github users API into recycler view

#### How should this be manually tested?
* Clone this repo: 
`$ git clone -b ft-handle-configuration-changes-156005187 https://github.com/leonardnjura/github-users`
* Open cloned directory `github-users` in Android Studio
* Build with Gradle
* Navigate to these directories:
`app/src/main/java/com/example/githubusers/adapter/`
`app/src/main/java/com/example/githubusers/view/`
* Run app in emulator device or actual connected device i.e. a physical Android phone or tablet
* Inspect Logcat with filter TAG: `MainActivity`

#### What are the relevant pivotal tracker stories?
[#156005187](https://www.pivotaltracker.com/story/show/156005187)

#### Screenshots
ezgif com-video-to-gif (3)

Screenshot 2019-03-13 at 11 16 33

### CODE REVIEW

File: app/src/main/java/com/example/githubusers/presenter/GithubUsersPresenter.java [hunk 1]
```diff
@@ -27,6 +27,14 @@ public GithubUsersPresenter(UsersParentView view, GithubService service) {
         }
     }
 
+//    public GithubUsersPresenter(UsersParentView view) {
```
Comment (johngorithm) [inline app/src/main/java/com/example/githubusers/presenter/GithubUsersPresenter.java]: Remove `commented out` lines of code

File: app/src/main/java/com/example/githubusers/service/GithubAPI.java [hunk 2]
```diff
@@ -7,6 +7,6 @@
 
 
 public interface GithubAPI {
-    @GET("/search/users?q=+language:java+location:nairobi&per_page=100")
+    @GET("/search/users?q=+language:java+location:moscow&per_page=100")
     Call<GithubUsersResponse> getUsers();
```
Comment (johngorithm) [inline app/src/main/java/com/example/githubusers/service/GithubAPI.java]: Nairobi is the specified location query parameter, kindly fix that 😄

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <resources>
-    <string name="user_list_message">Users in Kenya!</string>
+    <string name="user_list_message">Users in Russia!</string>
```
Comment (johngorithm) [inline app/src/main/res/values/strings.xml]: Let's return to Nairobi please

**Review by johngorithm — CHANGES_REQUESTED** [review]
Good Job, just some changes 👍

File: app/src/main/java/com/example/githubusers/service/GithubAPI.java (hunk 2 above)
Comment (leonardnjura) [reply app/src/main/java/com/example/githubusers/service/GithubAPI.java]: On it

**Review by johngorithm — APPROVED** [review]

**Comment by johngorithm** [comment]
LGTM.

@leonardnjura, I am merging this.

**Pull request merged**